### PR TITLE
WIP: limit TInt to the maximum value on *any* arch

### PR DESCRIPTION
**Pull request by karknu**

### CODE REVIEW

**Comment by karknu** [comment]
For entertainment purposes only as the network magic used by mainnet is larger than 2^29-1.

**Comment by karknu** [comment]
Implements #2186

File: ouroboros-network/src/Ouroboros/Network/NodeToClient.hs [hunk 1]
```diff
@@ -288,13 +290,23 @@ nodeToClientCodecCBORTerm = CodecCBORTerm {encodeTerm, decodeTerm}
...
+           then CBOR.TInt     (fromIntegral $ unNetworkMagic networkMagic)
+           else CBOR.TInteger (fromIntegral $ unNetworkMagic networkMagic)
+
+      decodeTerm (CBOR.TInt x) | x >= 0 && x <= maxInt32 = Right (NodeToClientVersionData $ NetworkMagic $ fromIntegral x)
+                               | otherwise    = Left $ T.pack $ "networkMagic out of bound: " <> show x
+      decodeTerm (CBOR.TInteger x) | x >= 0 && x <= maxWord32 = Right (NodeToClientVersionData $ NetworkMagic $ fromIntegral x)
+                               | otherwise                    = Left $ T.pack $ "networkMagic out of bound: " <> show x
       decodeTerm t             = Left $ T.pack $ "unknown encoding: " ++ show t
 
+      maxInt32 :: Integral a => a
```
Comment (coot) [inline ouroboros-network/src/Ouroboros/Network/NodeToClient.hs:303]: Do we need `maxInt32` and `maxWord32` to be polymorphic?

Comment (karknu) [reply ouroboros-network/src/Ouroboros/Network/NodeToClient.hs:303]: I don't remember.

**Comment by karknu** [comment]
We don't to support 32bit archs right now.

**Pull request closed without merging**